Saving bookmarks in the `zim-split` splitted ZIM files are crashing the application, when there is a 0 size extra zim chunk.

**Describe the bug**
Saving bookmarks in the `zim-split` splitted ZIM files are crashing the application, when there is a 0 size extra zim chunk.

**Expected behavior**
It should not crash.

**Steps to reproduce the behavior:**
1. Try to open any splitted ZIM file in the reader.
2. Click on the bookmark icon at the bottom.
3. You will see the crash.

**Environment**
- Version of Kiwix Android : 3.10.1
- Device : Google pixel 7a
- OS version : Android 13

**Logs**
```kotlin
 Process: org.kiwix.kiwixmobile, PID: 17268
                                                                                                    java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:562)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:878)
                                                                                                    Caused by: java.lang.reflect.InvocationTargetException
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:878) 
                                                                                                    Caused by: java.lang.Exception: Corrupted block detected
                                                                                                    	at org.kiwix.libkiwix.Book.update(Native Method)
                                                                                                    	at org.kiwix.kiwixmobile.core.main.CoreReaderFragment.toggleBookmark(CoreReaderFragment.kt:1758)
                                                                                                    	at org.kiwix.kiwixmobile.core.main.CoreReaderFragment_ViewBinding$6.doClick(CoreReaderFragment_ViewBinding.java:121)
                                                                                                    	at butterknife.internal.DebouncingOnClickListener.onClick(DebouncingOnClickListener.java:26)
                                                                                                    	at android.view.View.performClick(View.java:7729)
                                                                                                    	at android.view.View.performClickInternal(View.java:7706)
                                                                                                    	at android.view.View.-$$Nest$mperformClickInternal(Unknown Source:0)
                                                                                                    	at android.view.View$PerformClick.run(View.java:30484)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:959)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:100)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:232)
                                                                                                    	at android.os.Looper.loop(Looper.java:317)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8501)
```



https://github.com/kiwix/kiwix-android/assets/34593983/85176581-8617-4329-bd5a-4d11000b9648

--- a/core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreReaderFragment.kt
+++ b/core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreReaderFragment.kt
@@ -1752,32 +1752,38 @@ abstract class CoreReaderFragment :
   @Suppress("NestedBlockDepth")
   @OnClick(R2.id.bottom_toolbar_bookmark)
   fun toggleBookmark() {
-    getCurrentWebView()?.url?.let { articleUrl ->
-      zimReaderContainer?.zimFileReader?.let { zimFileReader ->
-        val libKiwixBook = Book().apply {
-          update(zimFileReader.jniKiwixReader)
-        }
-        if (isBookmarked) {
-          repositoryActions?.deleteBookmark(libKiwixBook.id, articleUrl)
-          snackBarRoot?.snack(R.string.bookmark_removed)
-        } else {
-          getCurrentWebView()?.title?.let {
-            repositoryActions?.saveBookmark(
-              LibkiwixBookmarkItem(it, articleUrl, zimFileReader, libKiwixBook)
-            )
-            snackBarRoot?.snack(
-              stringId = R.string.bookmark_added,
-              actionStringId = R.string.open,
-              actionClick = {
-                goToBookmarks()
-                Unit
-              }
-            )
+    try {
+      getCurrentWebView()?.url?.let { articleUrl ->
+        zimReaderContainer?.zimFileReader?.let { zimFileReader ->
+          val libKiwixBook = Book().apply {
+            update(zimFileReader.jniKiwixReader)
+          }
+          if (isBookmarked) {
+            repositoryActions?.deleteBookmark(libKiwixBook.id, articleUrl)
+            snackBarRoot?.snack(R.string.bookmark_removed)
+          } else {
+            getCurrentWebView()?.title?.let {
+              repositoryActions?.saveBookmark(
+                LibkiwixBookmarkItem(it, articleUrl, zimFileReader, libKiwixBook)
+              )
+              snackBarRoot?.snack(
+                stringId = R.string.bookmark_added,
+                actionStringId = R.string.open,
+                actionClick = {
+                  goToBookmarks()
+                  Unit
+                }
+              )
+            }
           }
         }
       } ?: kotlin.run {
         requireActivity().toast(R.string.unable_to_add_to_bookmarks, Toast.LENGTH_SHORT)
       }
+    } catch (ignore: Exception) {
+      // Catch the exception while saving the bookmarks for splitted zim files.
+      // we have an issue with split zim files, see #3827
+      requireActivity().toast(R.string.unable_to_add_to_bookmarks, Toast.LENGTH_SHORT)
     }
   }
 

